test/SparqlParserTest.cpp

```diff
@@ -335,20 +340,245 @@ SELECT ?a ?b ?c WHERE {
+      ASSERT_EQ(2u, pq._rootGraphPattern._children.size());
+
+      const auto& c = pq.children()[0].getBasic();
+      ASSERT_EQ(1u, c._whereClauseTriples.size());
+      ASSERT_EQ(0, pq._rootGraphPattern._filters.size());
+      ASSERT_EQ(3u, pq._offset);
+
+      ASSERT_EQ(c._whereClauseTriples[0]._s, "?movie");
+      ASSERT_EQ(c._whereClauseTriples[0]._p._iri, "<directed-by>");
+      ASSERT_EQ(c._whereClauseTriples[0]._o, "<Scott%2C%20Ridley>");
+
+      ASSERT_EQ(20u, pq._limit);
+      ASSERT_EQ(true, pq._orderBy[0]._desc);
+      ASSERT_EQ("?movie", pq._orderBy[0]._key);
+
+      auto sc = get<ParsedQuery::SelectClause>(pq._clause);
+      ASSERT_EQ(true, sc._distinct);
+      ASSERT_EQ(true, sc._varsOrAsterisk.isAsterisk());
+
+      vector<string> vvars = {"?movie", "?director", "?year"};
+      // TODO: ASSERT_ERROR
+      // ASSERT_EQ(vvars, sc._varsOrAsterisk.orderedVariablesFromQueryBody());
+
+      // -- SubQuery
+      // const auto& c_subquery = pq._rootGraphPattern._children[1].getBasic();
+      auto parsed_sub_query = get<GraphPatternOperation::Subquery>(
+          pq._rootGraphPattern._children[1].variant_);
+      const auto& c_subquery = get<GraphPatternOperation::BasicGraphPattern>(
+          parsed_sub_query._subquery._rootGraphPattern._children[0].variant_);
+      ASSERT_EQ(2u, c_subquery._whereClauseTriples.size());
+      ASSERT_EQ(1u,
+                parsed_sub_query._subquery._rootGraphPattern._filters.size());
+      ASSERT_EQ("?year",
+                parsed_sub_query._subquery._rootGraphPattern._filters[0]._lhs);
+      ASSERT_EQ("\"00-00-2000\"",
+                parsed_sub_query._subquery._rootGraphPattern._filters[0]._rhs);
+      ASSERT_EQ(SparqlFilter::GT,
+                parsed_sub_query._subquery._rootGraphPattern._filters[0]._type);
+      ASSERT_EQ(std::nullopt, parsed_sub_query._subquery._offset);
+
+      ASSERT_EQ(c_subquery._whereClauseTriples[0]._s, "?movie");
+      ASSERT_EQ(c_subquery._whereClauseTriples[0]._p._iri, "<directed-by>");
+      ASSERT_EQ(c_subquery._whereClauseTriples[0]._o, "?director");
+
+      ASSERT_EQ(c_subquery._whereClauseTriples[1]._s, "?movie");
+      ASSERT_EQ(c_subquery._whereClauseTriples[1]._p._iri, "<from-year>");
+      ASSERT_EQ(c_subquery._whereClauseTriples[1]._o, "?year");
+
+      ASSERT_EQ(std::nullopt, parsed_sub_query._subquery._limit);
+      ASSERT_EQ(true, parsed_sub_query._subquery._orderBy[0]._desc);
+      ASSERT_EQ("?director", parsed_sub_query._subquery._orderBy[0]._key);
+
+      auto sc_subquery =
+          get<ParsedQuery::SelectClause>(parsed_sub_query._subquery._clause);
+      ASSERT_EQ(false, sc_subquery._distinct);
+      ASSERT_EQ(false, sc_subquery._reduced);
+      ASSERT_EQ(true, sc_subquery._varsOrAsterisk.isAsterisk());
+      vector<string> vvars_subquery = {"?movie", "?director", "?year"};
+      ASSERT_EQ(vvars_subquery,
+                sc_subquery._varsOrAsterisk.orderedVariablesFromQueryBody());
+    }
+
```

Comment:
What is missing here:
```
SELECT * WHERE {
?a <b> ?c 
}
GROUP BY ?a ?c
```

Should give an error, because GROUP BY and SELECT * don't work together.